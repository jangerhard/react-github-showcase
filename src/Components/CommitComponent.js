import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

class CommitComponent extends React.Component {

    render() {

        const date = moment(this.props.committedDate).fromNow();

        return(
            <div style={{
                display: "flex",
                flexDirection: "row",
                justifyContent: "left",
                paddingLeft: "1em"
            }}>
                <div style={{
                    paddingLeft: "1em"
                }}>{this.props.abbreviatedOid}</div>
                :
                <div style={{
                    paddingLeft: "0.5em"
                }}>
                    {this.props.message}
                    </div>
                <div style={{
                    
                    paddingLeft: "0.5em"
                }}> </div>
                 ({date})
            </div>
        )
    }
}

CommitComponent.propTypes = {
    abbreviatedOid: PropTypes.string.isRequired,
    message: PropTypes.string.isRequired,
    committedDate: PropTypes.string.isRequired
};

export default CommitComponent;

