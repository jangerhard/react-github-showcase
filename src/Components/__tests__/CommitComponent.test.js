import ShallowRenderer from 'react-test-renderer/shallow';
import CommitComponent from "../CommitComponent";
import React from "react";

const abbreviatedOid = 'AAAAAA';
const message = 'This is a commit message';
const committedDate = '2018-05-04T19:59:54Z';

jest.mock('s-ago', () => () => ({ago: () => 'last week'}));

it('renders commit correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
        <CommitComponent
            message={message}
            abbreviatedOid={abbreviatedOid}
            committedDate={committedDate} />
    );
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
});