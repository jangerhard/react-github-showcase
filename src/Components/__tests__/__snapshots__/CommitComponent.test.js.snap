// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders commit correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingLeft": "1em",
    }
  }
>
  AAAAAA
  : 
  This is a commit message
  <div
    style={
      Object {
        "opacity": ".5",
        "paddingLeft": "0.5em",
      }
    }
  >
    Object {
      "ago": [Function],
    }
  </div>
</div>
`;
